src/azure-cli/azure/cli/command_modules/appservice/_validators.py

```diff
@@ -128,6 +128,18 @@ def validate_app_exists(cmd, namespace):
     if not app:
         raise ResourceNotFoundError("'{}' app not found in ResourceGroup '{}'".format(app, resource_group_name))
 
+def validate_functionapp_on_containerapp_vnet_add(cmd, namespace):
+    validate_functionapp_on_containerapp_vnet(cmd, namespace)
+    validate_add_vnet(cmd, namespace)
+
+def validate_functionapp_on_containerapp_vnet(cmd, namespace):
+    resource_group_name = namespace.resource_group_name
+    name = namespace.name
+    slot = namespace.slot
+    function_app = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get', slot)
+    if function_app.managed_environment_id is not None:
+        raise UnsupportOperationError('Config vnet for a function app on container app environment is not supported', 
+        'This function app is created in an App Environment. Go to App environment to configure Networking')
```

Comment:
I'm thinking we might want to use this function for other commands that we also need to block (e.g. az functionapp config command) in which case, I think it might be better if we rename this function. What do you think?